app/models/invenio_rdm_record_converter.rb

```diff
@@ -56,17 +56,52 @@ def versions(gf)
 
     def invenio_metadata(gf)
       {
-        "resource_type": resource_type(gf)
+        "resource_type": resource_type(gf.resource_type.shift),
+        "creators": creators(gf.creator)
       }
     end
 
-    def resource_type(gf)
-      irdm_subtype = DH_RESOURCE_TYPES[gf.resource_type.shift]
+    def resource_type(digitalhub_subtype)
+      irdm_subtype = DH_RESOURCE_TYPES[digitalhub_subtype]
       irdm_type = IRDM_RESOURCE_TYPES[irdm_subtype]
 
       {
         "type": irdm_type,
         "subtype": irdm_subtype
       }
     end
+
+    def creators(creators)
+      creators.map do |creator|
+        # dh_user = @known_creators.include?(creator) || (@known_creators << User.find_by(formal_name: creator))
+        # Galter Health Sciences Library is a User on production DigitalHub, but it does NOT have a formal_name value
+        dh_user = User.find_by(formal_name: creator)
+
+        if dh_user.present?
+          creator_type = "personal"
+          dh_user_display_name = dh_user.display_name.split(' ') # split name into components to be reused
+          family_name = dh_user_display_name.pop # remove last value from display name
+          given_name = dh_user_display_name.join(' ') # the remaining strings becomes given name
+          identifiers = dh_user.orcid.present? ? {"orcid": dh_user.orcid.split('/').pop} : {}
+          affiliations = [] # TODO: user.affiliation is nil for all users on production... where do we get this value?
+          # {name: "", identifiers: {"<scheme>": "<value>"}}
+        else
+          creator_type = "organisational"
+          family_name = ""
+          given_name = ""
+          identifiers = {} # TODO: can an organization have an orcid id?
+          affiliations = [] # only used for creator_type "personal"
+        end
+
+        {
+          "name": dh_user.formal_name, # will always be this regardless of type
+          "type": creator_type,
+          "role": "creator", # TODO: roles are from custom controlled vocab, for now use this
+          "given_name": given_name, # TODO: seemingly the international standard is first and middle, but America seems to just use first
+          "family_name": family_name,
+          "identifiers": identifiers, # TODO: other than ORCID what identifiers do we need to support? can we lookup users for identifiers on import?
```

Comment:
DigitalHub just has orcid from the list of supported creator identifier schemes in invenioRDM.

Reply:
That's what it seemed like to me. There were not a ton of users with that ID listed though and it seems to be an optional field for a user's profile. Maybe on the invenio import side of things it will make sense to try and look up the desired identifiers based on professional email or something?